### DISCUSSION TITLE
input .stl file from CAD design into BemRosetta

### DISCUSSION THREAD

**Discussion by Brizzum** [BEMRosetta/BEMRosetta#17] [Q&A] [answered]
Hello, I have several questions about inputing a CAD design into BemRosetta.
To better understand this question, my objective is to calculate hydrodynamic coefficients in NEMOH of a semisub structre I have created using CAD suftware. NEMOH requires only the the .dat file of the mesh of the hull which is under the mean water level (z=0, in BemRosetta).

This said, I find trouble in the sense that It is not easy to calculate the draft/freeboard of the depth of the hull, therefore If I have to take z=0 to be the water line, I do not know how to make my design such that I place correctly the length of the structure which is above the z=0 (freebaord) and under z=0 (draft).  

Another question I have is: how does BemRosetta process the material of the structure. I mean you can only input a .stl CAD exported file. However the mass that appears in the BemRosetta summary does not match with the one I get while looking at the Iproperties of the model. 

Finally I am having trouble when processing the mesh as it does not let me make any manual changes in the panels.

These questions might sound silly, but it is the first time getting around these types of softwares.
Any aid is welcome, thank you.

**Comment by izabala123** [comment, marked as answer]
Hi Brizzum

I will try to answer you:
- The .stl files don't include neither mass nor distance units. This way, BEMRosetta must 'guess' if it is mm or m, and considers that the mass is the submerged volume, that is, the volume under z=0.
- To get the right draft, I would load in BEMRosetta the full mesh including dry surface, then in 'Process' set the real mass and click 'Mass' button, then in 'Move' click on 'Archimede', to move the platform to be stable
- When you Export the file, click before on 'Underwater' to save just the wet mesh.
- Don't forget to set before the right density and gravity in 'Options'
- BEMRosetta does not include a mesh editor to let you change a mesh by hand. It includes tools to create basic meshes, and to try to heal, orient, and cut panels.

Best regards
Iñaki
